### PR TITLE
RBAC for rest of the API endpoints

### PR DESCRIPTION
**Pull request by Kami**
This pull request implements RBAC for rest of the API endpoints which were missing it:

- [x] Policy type (get all, get one)
- [x] Policy (get all, get one, create, update, delete)
- [x] Action entry_point view
- [x] Action parameters view
- [x] Rule views
- [x] Introduce new ephemeral "Timer" model which is only used for permission grants (same as we do for webhooks)

### CODE REVIEW

**Comment by Kami** [comment]
I introduced a new "TimerDB" meta model and resource type which is used to grant permissions on timers.

This way it's consistent with how we handle webhooks. Before that change, user needed to grant permissions on Triggers which is confusing and not really obvious.

Another small future improvement would be to allow users to specify optional "name" attribute (unique) for each timer rule trigger. This would make it a bit more user-friendly when working with timers and granting permissions on them. Right now user needs to use a system generation name which is not that user-friendly.

File: st2common/st2common/rbac/types.py [hunk 1]
```diff
@@ -91,6 +94,7 @@ class PermissionType(Enum):
     EXECUTION_RE_RUN = 'execution_rerun'
     EXECUTION_STOP = 'execution_stop'
     EXECUTION_ALL = 'execution_all'
+    EXECUTION_VIEWS_FILTERS_LIST = 'execution_view_filters_list'
```
Comment (enykeev) [inline st2common/st2common/rbac/types.py]: `EXECUTION_FILTERS_LIST` seems sufficiently unique

Comment (Kami) [inline st2common/st2common/rbac/types.py]: Yeah, was thinking about it quite a bit, but I prefer to be explicit :P

"pack health" vs "pack index health" - first one is a bit confusing and not really clear, imo.

Comment (enykeev) [inline st2common/st2common/rbac/types.py]: is it a problem with the naming or the concept of index health by itself?

Comment (Kami) [inline st2common/st2common/rbac/types.py]: Naming - pack health would imply something about pack health while pack index health implies something about pack index health (which is more correct).

Comment (enykeev) [inline st2common/st2common/rbac/types.py]: I'm saying that `VIEW` part is not needed, not that we should remove `INDEX` part. The whole idea of views was added out of sheer necessity due to limitations of pecan we no longer have. That's minor enough, yet kinda annoying.

File: st2common/st2common/rbac/types.py [hunk 2]
```diff
@@ -438,6 +491,8 @@ class SystemRole(Enum):
     PermissionType.EXECUTION_STOP: 'Ability to stop (cancel) a running execution.',
     PermissionType.EXECUTION_ALL: ('Ability to perform all the supported operations on a '
                                    'particular execution.'),
+    PermissionType.EXECUTION_VIEWS_FILTERS_LIST: ('Ability vew all the distinct execution '
```
Comment (enykeev) [inline st2common/st2common/rbac/types.py]: typo

File: st2common/st2common/rbac/types.py [hunk 3]
```diff
@@ -468,6 +523,10 @@ class SystemRole(Enum):
     PermissionType.WEBHOOK_ALL: ('Ability to perform all the supported operations on a particular '
                                  'webhook.'),
 
+    PermissionType.TIMER_LIST: 'Ability to list (view all) timers.',
```
Comment (enykeev) [inline st2common/st2common/rbac/types.py]: inconsistency

File: st2common/st2common/rbac/types.py (hunk 1 above)
Comment (Kami) [inline st2common/st2common/rbac/types.py]: Ah, I see.

Will think about it a bit, from a quick glance I agree with you, but I originally went with "view" for explicitness sake.

File: st2common/st2common/rbac/types.py (hunk 2 above)
Comment (Kami) [inline st2common/st2common/rbac/types.py]: Good catch, will fix.

**Pull request merged**